### ISSUE TITLE
Parametric regression models may have been misreporting results

### ISSUE THREAD

**Issue by CamDavidsonPilon** [CamDavidsonPilon/lifelines#931]
TLDR: if you have been using a __custom__ parametric regression model, or `GeneralizedGammaRegressionFitter`, you should update your code ASAP. Users of parametric AFT models are fine. 

----
It was discovered today that the parameters estimated in a parametric regression model could be not associated to the correct label (i.e. variable name). In practice, when a user hit `print_summary`, they would be displayed with the  correct estimates but with the wrong label beside it. This problem would also extend to any prediction or plotting methods. 

The root problem was a reordering of a dictionary in the autograd `flatten` function. Ex:
```
from autograd.misc import flatten
d = {'b': 1., 'a': 2., 'c': 0.}
array, unflatten = flatten(d)
print(array)
# ordered like a, b, c
```

lifelines _should_ have performed an `unflatten` on the result to recover the original ordering, but it didn't. 

It was first discovered in the documentation's example of the three parameter Cure model (of which there are no tests, since it was an example). Thankfully, almost all lifelines _implemented_ parametric regression fitters have tests against other survival libraries, and they are fine. However, `GeneralizedGammaRegressionFitter` does not have tests against other libraries, and could be reporting incorrect values.

**Comment by CamDavidsonPilon** [comment]
I've added some peripheral tests, and an `assert` in the code, that should prevent this from happening again.

**Comment by CamDavidsonPilon** [comment]
I think this problem has reappeared but now with the `.variance_matrix_`, which will effect the std. errors reported.

**Comment by StefanoBandera123** [comment]
Hi Cameron, I am running your example on customer churn https://github.com/CamDavidsonPilon/lifelines/blob/master/examples/Customer%20Churn.ipynb .

It seems I get this type of error when I call the fitter:
cph = CoxPHFitter().fit(churn_data, 'tenure', 'Churn', strata=strata_cols)

"ValueError: could not convert string to float: 'Female' "

How shoud I change the code?
